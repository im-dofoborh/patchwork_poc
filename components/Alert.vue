<template>
  <div
    v-show="show"
    :class="`sg-alert sg-alert--${type}`">
    <button
      class="sg-alert__button"
      @click="show = false">
      Ã—
    </button>
    <small
      v-html="message" />
  </div>
</template>

<script>
  export default {
    props: {
      message: {
        type: String,
        required: true,
      },
      type: {
        type: String,
        default: 'info',
      },
    },

    data() {
      return {
        show: true,
      }
    },
  }
</script>

<style lang="scss" scoped>
  .sg-alert {
    color: $color--form--info--regular;
    background: $color--form--warning--light;
    display: block;
    padding: spacing-unit(sm);
    position: relative;

    &--error {
      color: $color--form--warning--regular;
      background: $color--form--warning--light;
    }

    &--success {
      color: $color--form--success--regular;
      background: $color--form--success--light;
    }

    &__button {
      border: 0;
      background: transparent;
      margin-left: auto;
      position: absolute;
      color: rgba(0, 0, 0, 0.5);
      font-weight: bold;
      padding: spacing-unit(xxs) spacing-unit(sm);
      right: 0;
      top: 0;
    }
  }
</style>