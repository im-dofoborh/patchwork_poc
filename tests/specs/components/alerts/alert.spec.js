import { mount, createLocalVue } from '@vue/test-utils';
import Alert from '~/components/Alert';

describe('Alert component', () => {
  let localVue = null;
  let wrapper = null;

  beforeEach(() => {
    localVue = createLocalVue();
    localVue.component('alert', Alert);
    wrapper = mount(Alert, {
      localVue,
      propsData: {
        message: 'Look out, this is an alert',
      }
    })
  })

  it('renders default variant', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders alternative type variant', () => {
    wrapper.setProps({
      type: 'error',
    })
    expect(wrapper.html()).toMatchSnapshot();
  });
});
