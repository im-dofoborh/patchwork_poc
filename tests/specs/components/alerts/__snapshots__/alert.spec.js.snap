// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert component renders alternative type variant 1`] = `
<div class="sg-alert sg-alert--error"><button class="sg-alert__button">
    ×
  </button> <small>Look out, this is an alert</small></div>
`;

exports[`Alert component renders default variant 1`] = `
<div class="sg-alert sg-alert--info"><button class="sg-alert__button">
    ×
  </button> <small>Look out, this is an alert</small></div>
`;
